lib/jellyfish/component/hls/request_handler.ex

```diff
@@ -162,13 +188,24 @@ defmodule Jellyfish.Component.HLS.RequestHandler do
     end
   end
 
-  defp update_status(status, last_partial) do
+  defp wait_for_partial_ready(room_id, filename) do
+    GenServer.cast(registry_id(room_id), {:preload_hint, room_id, filename, self()})
+
+    receive do
+      :preload_hint_ready ->
+        EtsHelper.get_partial(room_id, filename)
+    end
+  end
+
+  defp update_status(status, last_partial, preload_hints) do
```

Comment:
I'd argue for renaming this function, as it currently has a side-effect of notifying pids that are waiting for partial/preload hint. Maybe something like `update_status_and_notify_waiting_pids` but shorter? Maybe even extract the notifying to another function?